pkg/controller/clusterinstallation/mattermost.go

```diff
@@ -288,28 +284,19 @@ func (r *ReconcileClusterInstallation) updateMattermostDeployment(mi *mattermost
 		}
 	}
 
-	// Ensure deployment replicas is the same as the spec
-	if *d.Spec.Replicas != mi.Spec.Replicas {
-		reqLogger.Info("Updating mattermost deployment replica count")
-		d.Spec.Replicas = &mi.Spec.Replicas
-		update = true
-	}
-
-	updatedLabels := ensureLabels(mattermostv1alpha1.ClusterInstallationLabels(mi.Name), d.Labels)
-	if !reflect.DeepEqual(updatedLabels, d.Labels) {
-		reqLogger.Info("Updating mattermost deployment labels")
-		update = true
-		d.Labels = updatedLabels
+	patchResult, err := objectMatcher.DefaultPatchMaker.Calculate(original, modified)
+	if err != nil {
+		reqLogger.Error(err, "Error checking the difference in the deployment")
+		return err
 	}
 
-	if update {
-		mu := intstr.FromInt(int(mi.Spec.Replicas - 1))
```

Comment:
As noted in the comment for this method, this code would set the max unavailable value of the pod deployment to a specific value based on the replicas. The idea was to always have a sane max unavailable value even though one may not be set directly in the original deployment object.

I am only 1/5 on keeping this intact, but just note that if we remove this then we will change the rolling update behavior. Also, we should update the comment for this method so it no longer includes the max unavailable line.

Reply:
we can set in the deployment this should not change. i will do the logic there.